lib/src/signed_video_h26x_auth.c

```diff
@@ -720,6 +721,41 @@ prepare_for_validation(signed_video_t *self)
   return status;
 }
 
+// If public_key is not received then try to decode all recurrent tags.
+static bool
+is_recurrent_data_decoded(signed_video_t *self)
+{
+  gop_state_t *gop_state = &(self->gop_state);
+  gop_info_detected_t *gop_info_detected = &(self->gop_info_detected);
+  signed_video_latest_validation_t *latest = self->latest_validation;
+  h26x_nalu_list_t *nalu_list = self->nalu_list;
+
+  if (!self->has_public_key && gop_state->signing_present) {
```

Comment:
Since the whole function is sort of wrapped in within this if-statement I suggest turning it into a sanity check.